From 33869db394762f1edf85f541a6c61d162c13c0cc Mon Sep 17 00:00:00 2001
From: Cesar Strauss <cstrauss@cesar.(none)>
Date: Fri, 24 Oct 2008 22:14:59 -0200
Subject: [PATCH] MinGW does not have getlogin.

---
 configure.ac.in |    2 +-
 src/f_print.c   |    4 ++++
 2 files changed, 5 insertions(+), 1 deletions(-)

diff --git a/configure.ac.in b/configure.ac.in
index 6a3c02c..e701501 100644
--- a/configure.ac.in
+++ b/configure.ac.in
@@ -352,7 +352,7 @@ AC_C_CONST
 
 # Checks for library functions.
 AC_TYPE_SIGNAL
-AC_CHECK_FUNCS(getcwd strstr vsnprintf snprintf chown)
+AC_CHECK_FUNCS(getcwd strstr vsnprintf snprintf chown getlogin)
 
 # 
 # Checks for header files end
diff --git a/src/f_print.c b/src/f_print.c
index 1820544..ce4b584 100644
--- a/src/f_print.c
+++ b/src/f_print.c
@@ -114,7 +114,9 @@ int f_print_header(TOPLEVEL *toplevel, FILE *fp,
   fprintf(fp, "%%%%Creator: gEDA gschem %s\n"
 	  "%%%%CreationDate: %s"
 	  "%%%%Title: %s\n"
+#ifdef HAVE_GETLOGIN
 	  "%%%%Author: %s\n"
+#endif
 	  "%%%%BoundingBox: %d %d %d %d\n"
 	  "%%%%Orientation: %s\n"
 	  "%%%%Pages: 1\n"
@@ -123,7 +125,9 @@ int f_print_header(TOPLEVEL *toplevel, FILE *fp,
 	  DATE_VERSION,
 	  ctime(&current_time),
 	  toplevel->page_current->page_filename,
+#ifdef HAVE_GETLOGIN
 	  getlogin(),
+#endif
 	  llx, lly, urx, ury,
 	  ((toplevel->print_orientation == LANDSCAPE)
 	   ? "Landscape" : "Portrait")
-- 
1.5.4.3

