From 343d54becf789229bdc718dcf7a002a48817f01a Mon Sep 17 00:00:00 2001
From: Cesar Strauss <cstrauss@cesar.(none)>
Date: Fri, 24 Oct 2008 22:17:33 -0200
Subject: [PATCH] MinGW does not have sys/wait.h

---
 src/s_clib.c |    7 +++++++
 1 files changed, 7 insertions(+), 0 deletions(-)

diff --git a/src/s_clib.c b/src/s_clib.c
index f3057c7..1fa3ca1 100644
--- a/src/s_clib.c
+++ b/src/s_clib.c
@@ -123,7 +123,14 @@
 #include <dmalloc.h>
 #endif
 
+#ifdef HAVE_SYS_WAIT_H
 #include <sys/wait.h>
+#else
+#define WIFSIGNALED(x) 0
+#define WTERMSIG(x)    0
+#define WIFEXITED(x)   1
+#define WEXITSTATUS(x) 0
+#endif
 
 #include <time.h>
 
-- 
1.5.4.3

