From fd758bbfd66aacea304cb56d96904d2b865e5a4f Mon Sep 17 00:00:00 2001
From: Cesar Strauss <cstrauss@cesar.(none)>
Date: Fri, 24 Oct 2008 22:21:36 -0200
Subject: [PATCH] Use pkg-config support for Guile if present.

---
 configure.ac.in |   32 ++++++++++++++++++++++----------
 1 files changed, 22 insertions(+), 10 deletions(-)

diff --git a/configure.ac.in b/configure.ac.in
index e701501..9954fd8 100644
--- a/configure.ac.in
+++ b/configure.ac.in
@@ -97,9 +97,28 @@ IT_PROG_INTLTOOL(0.35.0)
 # Check for guile start
 #
 
-# Find about the installed guile
-AC_GEDA_MACRO_CHECK([GUILE_FLAGS], [guile.m4])
-GUILE_FLAGS
+# Check for pkg-config
+AC_PATH_PROG(PKG_CONFIG, pkg-config, no)
+if test $PKG_CONFIG = no; then
+   AC_MSG_ERROR([Cannot find pkg-config, make sure it is installed and in your PATH])
+fi
+
+PKG_CHECK_MODULES(GUILE, guile-1.8, GUILE_PC="yes", no_GUILE_PC="yes")
+
+if test "$GUILE_PC" = "yes" 
+then
+
+   GUILE_LDFLAGS=$GUILE_LIBS
+   GUILE_VERSION=`$PKG_CONFIG guile-1.8 --modversion`
+
+else
+
+  # Find about the installed guile
+  AC_GEDA_MACRO_CHECK([GUILE_FLAGS], [guile.m4])
+  GUILE_FLAGS
+  GUILE_VERSION=`$GUILE_CONFIG info guileversion`
+
+fi
 
 # Check Guile version
 guile_need_major=1
@@ -107,7 +126,6 @@ guile_need_minor=6
 guile_need_version="$guile_need_major[].$guile_need_minor.0"
 
 AC_MSG_CHECKING([Guile version >= $guile_need_version])
-GUILE_VERSION=`$GUILE_CONFIG info guileversion`
 
 guile_major=`echo "$GUILE_VERSION" | sed 's/\([[^.]][[^.]]*\).*/\1/'`
 guile_minor=`echo "$GUILE_VERSION" | sed 's/[[^.]][[^.]]*.\([[^.]][[^.]]*\).*/\1/'`
@@ -177,12 +195,6 @@ fi
 # Check for gtk+ 2.4 start
 # 
 
-# Check for pkg-config
-AC_PATH_PROG(PKG_CONFIG, pkg-config, no)
-if test $PKG_CONFIG = no; then
-   AC_MSG_ERROR([Cannot find pkg-config, make sure it is installed and in your PATH])
-fi
-
 PKG_CHECK_MODULES(GTK24, gtk+-2.0 >= 2.4.0, GTK24="yes", no_GTK24="yes")
 
 # This next bit of code figures out what gtk we need to use.
-- 
1.5.4.3

