From 7d1e495d9b98f95c8b18c61799d10d7ac72ad2bc Mon Sep 17 00:00:00 2001
From: Cesar Strauss <cstrauss@cesar.(none)>
Date: Fri, 24 Oct 2008 22:26:25 -0200
Subject: [PATCH] MinGW does not have realpath.

---
 src/f_basic.c |    5 ++++-
 1 files changed, 4 insertions(+), 1 deletions(-)

diff --git a/src/f_basic.c b/src/f_basic.c
index cfaf94b..2a8da4d 100644
--- a/src/f_basic.c
+++ b/src/f_basic.c
@@ -479,9 +479,12 @@ char* f_normalize_filename(const gchar *filename)
     return NULL;
   }
 
+#ifndef __MINGW32__
   realpath(filename, filename_buffer);  /* places reult in filename_buffer */
   full_filename = g_strdup (filename_buffer);
-
+#else
+  full_filename = g_strdup (filename);
+#endif
 #ifdef DEBUG
   printf("In f_normalize_filename, returning full_filename= %s \n", full_filename);
 #endif 
-- 
1.5.4.3

