diff -ur a/libguile/fports.c b/libguile/fports.c
--- a/libguile/fports.c	2008-11-02 18:20:23.000000000 +0000
+++ b/libguile/fports.c	2008-11-02 18:20:50.000000000 +0000
@@ -414,8 +414,13 @@
       else if ((fdes == fileno (stdout) || fdes == fileno (stderr)) && 
 	       isatty (fdes))
 	flags = O_WRONLY;
-      else
-	flags = buf.st_mode;
+      else if ((buf.st_mode & (_S_IREAD | _S_IWRITE)) ==
+                              (_S_IREAD | _S_IWRITE))
+        flags = O_RDWR;
+      else if (buf.st_mode & _S_IREAD)
+        flags = O_RDONLY;
+      else if (buf.st_mode & _S_IWRITE)
+        flags = O_WRONLY;
     }
   return flags;
 }
