--- a/ice-9/boot-9.scm	2008-11-02 18:20:23.000000000 +0000
+++ b/ice-9/boot-9.scm	2008-11-02 18:20:50.000000000 +0000
@@ -1625,9 +1625,12 @@
        (apply basic-load
 	      (if (and oldname
 		       (> (string-length filename) 0)
-		       (not (char=? (string-ref filename 0) #\/))
+                       (not (or (and (> (string-length filename) 1)
+                                     (char=? (string-ref filename 1) #\:))
+		                (char=? (string-ref filename 0) #\/)
+                                (char=? (string-ref filename 0) #\\)))
 		       (not (string=? (dirname oldname) ".")))
-		  (string-append (dirname oldname) "/" filename)
+		  (string-append (dirname oldname) "\\" filename)
 		  filename)
 	      reader)))))
 
