diff --git a/src/globals.c b/src/globals.c
index 1baeaf7..40819a4 100644
--- a/src/globals.c
+++ b/src/globals.c
@@ -44,16 +44,16 @@ int do_logging=TRUE;
 int logging_dest=STDOUT_TTY;
 
 /* these are required by libgeda */
-void (*arc_draw_func)() = o_arc_recalc;
-void (*box_draw_func)() = o_box_recalc;
-void (*picture_draw_func)() = o_picture_recalc;
-void (*circle_draw_func)() = o_circle_recalc;
-void (*complex_draw_func)() = o_complex_recalc;
-void (*line_draw_func)() = o_line_recalc;
-void (*net_draw_func)() = o_net_recalc;
-void (*text_draw_func)() = o_text_recalc;
-void (*bus_draw_func)() = o_bus_recalc;
-void (*pin_draw_func)() = o_pin_recalc;
+void (*arc_draw_func)() = NULL;
+void (*box_draw_func)() = NULL;
+void (*picture_draw_func)() = NULL;
+void (*circle_draw_func)() = NULL;
+void (*complex_draw_func)() = NULL;
+void (*line_draw_func)() = NULL;
+void (*net_draw_func)() = NULL;
+void (*text_draw_func)() = NULL;
+void (*bus_draw_func)() = NULL;
+void (*pin_draw_func)() = NULL;
 void (*select_func)() = NULL;
 void (*x_log_update_func)() = s_log_update;
 int (*load_newer_backup_func)() = NULL;

