diff --git a/src/globals.c b/src/globals.c
index e0d25a5..d5fc4a5 100644
--- a/src/globals.c
+++ b/src/globals.c
@@ -49,9 +49,6 @@ GdkColor black;
 
 int logging_dest = LOG_WINDOW;
 
-/* these are required by libgeda */
-void (*select_func)()        = o_select_object; /* NEW SELECTION code */
-
 /* command line options */
 int quiet_mode = FALSE;
 int verbose_mode = FALSE;
diff --git a/src/gschem.c b/src/gschem.c
index 633d33d..9b3384a 100644
--- a/src/gschem.c
+++ b/src/gschem.c
@@ -178,6 +178,7 @@ void main_prog(void *closure, int argc, char *argv[])
   pin_draw_func = o_pin_draw;
   text_draw_func = o_text_draw;
   load_newer_backup_func = x_fileselect_load_backup;
+  select_func = o_select_object;
 
   /*! \todo Probably the file name shuold be defined elsewhere */
   /* create log file right away even if logging is enabled */

