--- a/configure.ac	2011-03-27 22:36:27.417298518 -0300
+++ b/configure.ac	2011-03-27 22:40:50.732063466 -0300
@@ -345,20 +345,20 @@ AC_ARG_WITH(png,dnl
 [  --with-png=DIR          where to find the png library],,
   withval=yes)
 
-# libpng12.so is a nightmare. We have to test separately for
-# libpng12-config and libpng-config. If we find libpng12-config,
+# libpng14.so is a nightmare. We have to test separately for
+# libpng14-config and libpng-config. If we find libpng14-config,
 # we don't look for the other, which will just be a symlink to it,
 # confusing autoconf into thinking the library itself will be named -lpng,
 # which it won't be. Awful. This code will suffice unless the libpng
 # authors decide to do this AGAIN. Which I really hope they won't. TBB
 
 if test "$withval" != no; then
-  AC_PATH_PROG([LIBPNG12_CONFIG], [libpng12-config])
+  AC_PATH_PROG([LIBPNG14_CONFIG], [libpng14-config])
   AC_PATH_PROG([LIBPNG_CONFIG], [libpng-config])
-  if test -n "$LIBPNG12_CONFIG"; then
-    libpng_CPPFLAGS=`libpng12-config --cflags`
+  if test -n "$LIBPNG14_CONFIG"; then
+    libpng_CPPFLAGS=`libpng14-config --cflags`
     # should be --ldopts, but it's currently broken
-    libpng_LDFLAGS=`libpng12-config --ldflags`
+    libpng_LDFLAGS=`libpng14-config --ldflags`
     libpng_LDFLAGS=`echo " $libpng_LDFLAGS" | sed 's/ -l[[^ ]][[^ ]]*//g'`
   elif test -n "$LIBPNG_CONFIG"; then
     libpng_CPPFLAGS=`libpng-config --cflags`
@@ -378,9 +378,9 @@ if test "$withval" != no; then
   _ldflags="$LDFLAGS"
   LDFLAGS="$libpng_LDFLAGS $LDFLAGS"
 
-  if test -n "$LIBPNG12_CONFIG"; then
-    AC_CHECK_LIB(png12,png_create_read_struct,
-      [LIBS="-lpng12 $LIBS"
+  if test -n "$LIBPNG14_CONFIG"; then
+    AC_CHECK_LIB(png14,png_create_read_struct,
+      [LIBS="-lpng14 $LIBS"
        FEATURES="GD_PNG $FEATURES"
        AC_DEFINE(HAVE_LIBPNG, 1, [ Define if you have the png library. ])],
       [LDFLAGS="$_ldflags"  
@@ -405,9 +405,9 @@ else
   AC_MSG_WARN([libpng is required - see http://www.libpng.org/pub/png/])
 fi
 
-#TBB: png12's png_create_read_struct is just as good 
+#TBB: png14's png_create_read_struct is just as good 
 #2.0.23: should be = not ==  
-if test "$ac_cv_lib_png12_png_create_read_struct" = yes; then
+if test "$ac_cv_lib_png14_png_create_read_struct" = yes; then
   ac_cv_lib_png_png_create_read_struct=yes
 fi 
 
