--- /home/pcjc2/gedasrc/git/gschem/src/o_text.c	2010-01-18 22:32:08.320739809 +0000
+++ geda-gaf/gschem/src/o_text.c	2010-01-18 22:32:33.891683442 +0000
@@ -159,7 +159,6 @@
   PangoAttrList *attrs;
   cairo_font_options_t *options;
   double font_size_pt;
-  char *font_string;
   char *unescaped;
 
   context = pango_layout_get_context (layout);
@@ -174,9 +173,10 @@
   pango_cairo_context_set_resolution (context, 1000. * scale_factor);
   font_size_pt = o_text_get_font_size_in_points (w_current->toplevel,
                                                  o_current);
-  font_string = g_strdup_printf ("%s %f\n", FONT_NAME, font_size_pt);
-  desc = pango_font_description_from_string (font_string);
-  g_free (font_string);
+
+  desc = pango_font_description_new ();
+  pango_font_description_set_family_static (desc, FONT_NAME);
+  pango_font_description_set_size (desc, (int)((double)PANGO_SCALE * font_size_pt));
 
   pango_layout_set_font_description (layout, desc);
   font_metrics = pango_context_get_metrics (context, desc, NULL);
