--- ../geda-gaf-1.6.0-2/geda-gaf-1.6.0/gschem/src/gschem.c	2010-01-19 15:40:33.546875000 -0200
+++ geda-gaf-1.6.0/gschem/src/gschem.c	2010-01-19 15:37:32.828125000 -0200
@@ -145,7 +145,7 @@ void main_prog(void *closure, int argc, 
   gtk_set_locale();
 
   /* This must be the same for all locales */
-  setlocale(LC_NUMERIC, "POSIX");
+  setlocale(LC_NUMERIC, "C");
 
   /* Disable gtk's ability to set the locale. */ 
   /* If gtk is allowed to set the locale, then it will override the     */
@@ -345,7 +345,7 @@ int main (int argc, char *argv[])
 
 #if ENABLE_NLS
   setlocale(LC_ALL, "");
-  setlocale(LC_NUMERIC, "POSIX");
+  setlocale(LC_NUMERIC, "C");
   bindtextdomain("geda-gschem", LOCALEDIR);
   textdomain("geda-gschem");
   bind_textdomain_codeset("geda-gschem", "UTF-8");
