diff --git a/configure.ac b/configure.ac
index 43ea2c0..e86d15c 100644
--- a/configure.ac
+++ b/configure.ac
@@ -50,6 +50,10 @@ AC_DEFINE_UNQUOTED(PCB_PATH_DELIMETER,"$PCB_PATH_DELIMETER",[Search path separat
 dnl Checks for programs.
 AC_PROG_CC
 
+if test "x$WIN32" = "xyes" ; then
+  AC_CHECK_TOOL([WINDRES], [windres])
+fi
+
 # i18n
 GETTEXT_PACKAGE=$PACKAGE
 AH_TEMPLATE([GETTEXT_PACKAGE], [Name of this program's gettext domain])
diff --git a/src/Makefile.am b/src/Makefile.am
index 32ec016..856d334 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -267,7 +267,7 @@ endif
 # desktop and application toolbar
 if WIN32
 pcb_icon.o : pcb_icon.ico $(srcdir)/hid/gtk/pcb.rc
-	windres $(srcdir)/hid/gtk/pcb.rc $@
+	$(WINDRES) $(srcdir)/hid/gtk/pcb.rc $@
 
 pcb_icon.ico: $(top_srcdir)/data/pcb_icon.ico
 	cp $(top_srcdir)/data/pcb_icon.ico $@
