diff --git a/src/globals.c b/src/globals.c
index 390a60f..6cf4bdb 100644
--- a/src/globals.c
+++ b/src/globals.c
@@ -43,7 +43,7 @@ int quiet_mode=FALSE;
  * for gattrib -- no drawing is done. */
 void (*arc_draw_func)() = NULL;
 void (*box_draw_func)() = NULL;
-void (*picture_draw_func)() = o_picture_recalc;
+void (*picture_draw_func)() = NULL;
 void (*circle_draw_func)() = NULL;
 void (*complex_draw_func)() = NULL;
 void (*line_draw_func)() = NULL;

