diff --git a/configure.ac b/configure.ac
index 4d7cb31..2349b10 100644
--- a/configure.ac
+++ b/configure.ac
@@ -158,6 +158,10 @@ AC_PROG_MAKE_SET
 AC_PROG_INSTALL
 AC_PROG_LIBTOOL
 
+if test "x$WIN32" = "xyes" ; then
+  AC_CHECK_TOOL([WINDRES], [windres])
+fi
+
 # if we have gcc then add -Wall
 if test "x$GCC" = "xyes"; then
 	if test -z "`echo "$CFLAGS" | grep "\-Wall" 2> /dev/null`" ; then
diff --git a/src/Makefile.am b/src/Makefile.am
index fca5141..47f0614 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -92,7 +92,7 @@ gerbv_DEPENDENCIES = libgerbv.la
 # desktop and application toolbar
 if WIN32
 gerbv_icon.o : gerbv_icon.ico $(srcdir)/gerbv.rc
-	windres $(srcdir)/gerbv.rc $@
+	$(WINDRES) $(srcdir)/gerbv.rc $@
 
 gerbv_icon.ico: $(top_srcdir)/desktop/gerbv_icon.ico
 	cp $(top_srcdir)/desktop/gerbv_icon.ico $@
