diff --git a/src/Makefile.am b/src/Makefile.am
index 47f0614..fe2cffa 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -72,7 +72,7 @@ libgerbv_la_SOURCES= \
 # 6. If any interfaces have been removed since the last public release, then
 #    set age to 0.
 #
-libgerbv_la_LDFLAGS = -version-info 1:1:0
+libgerbv_la_LDFLAGS = -version-info 1:1:0 -no-undefined
 
 gerbv_SOURCES = \
 		opdefines.h \
